### PR TITLE
Fix throwing non-promise-rejection errors.

### PR DESCRIPTION
**Pull request by luki-**

### CODE REVIEW

**Review by kubek2k — COMMENTED** [review]
I am not really convinced to putting packages locks in the projects TBH

**Comment by luki-** [comment]
I may remove it if you want me to.

File: index.js [hunk 1]
```diff
@@ -64,7 +64,7 @@ module.exports = (logFn, baseName) =>
                     })
                     .catch((err) => {
                         collected.measure[`${name}.error_duration`] = err.duration;
-                        throw err.reason;
+                        throw err.reason || err;
```
Comment (kubek2k) [inline index.js:67]: can you provide me an example when the err won't have reason? I look through timed function and it seems pretty stable in that. Providing a `null` as a promise arg?

Comment (luki-) [reply index.js:67]: Sure. We have (actually had, fixed in another PR) an issue that a non-promise object was provided as `promise` argument here. This caused an error like `[].then is not a function` being thrown that doesn't have a `reason` property.

Comment (kubek2k) [reply index.js:67]: got it 🙇 . Just wondering if validation of incoming object to be actual promise wouldn't be a better idea.

Comment (luki-) [reply index.js:67]: It could be a good idea but as an additional check (or maybe simply `Promise.resolve` given value).

Comment (kubek2k) [reply index.js:67]: do whatever you find sensible

**Comment by kubek2k** [comment]
please remove the lockfile

**Comment by luki-** [comment]
@kubek2k Updated.

**Review by kubek2k — APPROVED** [review]
🚢

**Comment by luki-** [comment]
Ref: https://github.schibsted.io/spt-mediaplatform/relate/issues/645

**Pull request merged**